src/NuGet.Core/NuGet.ProjectModel/PackageSpecOperations.cs

```diff
@@ -52,7 +52,12 @@ public static bool HasPackage(PackageSpec spec, string packageId)
 
             foreach (var list in lists)
             {
-                AddOrUpdateDependencyInDependencyList(list, dependency.Id, dependency.VersionRange);
+                AddOrUpdateDependencyInDependencyList(spec, list, dependency.Id, dependency.VersionRange);
+            }
+
+            foreach (IDictionary<string, CentralPackageVersion> centralPackageVersionList in GetCentralPackageVersionLists(spec, frameworksToAdd))
+            {
+                centralPackageVersionList[dependency.Id] = new CentralPackageVersion(dependency.Id, dependency.VersionRange);
```

Comment:
immutable types are awesome :D

Reply:
Haha nevermind you are right that its immutable.